test/conftest.py

```diff
@@ -46,6 +46,8 @@ def pytest_sessionfinish(session, exitstatus):
     # If we are the main worker thread, we've been called last, so do the cleanup.
     if worker_id is None:
         for runtime in CONTAINER_RUNTIMES:
+            if runtime not in FOUND_RUNTIMES:
+                continue
```

Comment:
This has nothing to do with invalid requirements parsing, so confusing why this is here.

Also, I'm fairly certain the reason I didn't use FOUND_RUNTIMES here when adding this code was to facilitate cleanup of images when running locally (the entire reason I added this callback) where you might have failures across different runtimes across multiple runs. Using CONTAINER_RUNTIMES makes sure we always check for left over images, even if we chose to skip a runtime at some point.

Reply:
Sorry this was a local change. Inadvertently got into this PR.